apps/scratch/src/pages/tests/Edit.spec.tsx

```diff
@@ -0,0 +1,142 @@
+          },
+        };
+      }),
+    );
+
+    await page.goto(`${baseURL!}/edit/some-task-id`);
+
+    await page.waitForSelector("#root");
+  });
+
+  test("can get height via window.postMessage", async ({ page }) => {
+    await page.evaluate(() => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const event = new Event("message") as any;
+
+      event.source = window.parent;
+      event.data = { id: 0, type: "request", procedure: "getHeight" };
+      window.dispatchEvent(event);
+    });
+
+    await page.waitForFunction(() => window.postedMessages.length > 0);
+
+    const messages = await page.evaluate(() => window.postedMessages);
+
+    expect(messages).toHaveLength(1);
+
+    expect(messages[0].message).toEqual({
+      id: 0,
+      type: "response",
+      procedure: "getHeight",
+      result: expect.any(Number),
+    });
+  });
+
+  test("can get task via window.postMessage", async ({ page }) => {
+    await page.evaluate(() => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const event = new Event("message") as any;
+
+      event.source = window.parent;
+      event.data = { id: 0, type: "request", procedure: "getTask" };
+      window.dispatchEvent(event);
+    });
+
+    await page.waitForFunction(() => window.postedMessages.length > 0);
+
+    const messages = await page.evaluate(() => window.postedMessages);
+
+    expect(messages).toHaveLength(1);
+
+    expect(messages[0].message).toEqual({
+      id: 0,
+      type: "response",
+      procedure: "getTask",
+      // blobs cannot be transferred, see https://github.com/puppeteer/puppeteer/issues/3722
+      result: {},
+    });
+  });
+
+  test("can load task via window.postMessage", async ({ page }) => {
+    page.route(/test-task.sb3$/, (route) =>
+      route.fulfill({
+        body: testTask,
+        contentType: "application/zip",
+        status: 200,
+      }),
+    );
+
+    await page.evaluate(async () => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const event = new Event("message") as any;
+
+      const task = await fetch("https://example.com/test-task.sb3").then(
+        (response) => response.blob(),
+      );
+
+      event.source = window.parent;
+      event.data = {
+        id: 0,
+        type: "request",
+        procedure: "loadTask",
+        arguments: task,
+      };
+      window.dispatchEvent(event);
+    });
+
+    await page.waitForFunction(() => window.postedMessages.length > 0);
+
+    const messages = await page.evaluate(() => window.postedMessages);
+
+    expect(messages).toHaveLength(1);
+
+    expect(messages[0].message).toEqual({
+      id: 0,
+      type: "response",
+      procedure: "loadTask",
+      result: undefined,
+    });
+
+    // ensure the block visibility is correctly loaded
+    expect(
+      page.locator(
+        "[data-id='motion_turnright'] [data-testid='shown-block-button']",
```

Comment:
Until we have a page object model, at least constants with explicit names here would help :)

Reply:
Oh yes of course! Sorry I thought this selector was very explicit ^^